tools/plugins/com.liferay.ide.upgrade.core/src/com/liferay/blade/upgrade/liferay70/apichanges/BaseLiferayDescriptorVersion.java

```diff
@@ -0,0 +1,131 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.blade.upgrade.liferay70.apichanges;
+
+import com.liferay.blade.api.AutoMigrateException;
+import com.liferay.blade.api.AutoMigrator;
+import com.liferay.blade.api.Problem;
+import com.liferay.blade.api.SearchResult;
+import com.liferay.blade.api.XMLFile;
+import com.liferay.blade.upgrade.XMLFileMigrator;
+
+import java.io.File;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.eclipse.core.resources.IFile;
+import org.eclipse.wst.sse.core.StructuredModelManager;
+import org.eclipse.wst.sse.core.internal.provisional.IModelManager;
+import org.eclipse.wst.xml.core.internal.document.DocumentTypeImpl;
+import org.eclipse.wst.xml.core.internal.provisional.document.IDOMDocument;
+import org.eclipse.wst.xml.core.internal.provisional.document.IDOMModel;
+
+/**
+ * @author Seiphon Wang
+ */
+@SuppressWarnings("restriction")
+public abstract class BaseLiferayDescriptorVersion extends XMLFileMigrator implements AutoMigrator {
+
+	public BaseLiferayDescriptorVersion(String publicIDPattern, String version) {
+		_publicIDPattern = publicIDPattern;
+		_version = version;
+	}
+
+	@Override
+	public int correctProblems(File file, List<Problem> problems) throws AutoMigrateException {
+		try {
+			IFile xmlFile = getXmlFile(file);
+
+			IModelManager modelManager = StructuredModelManager.getModelManager();
+
+			IDOMModel domModel = (IDOMModel)modelManager.getModelForRead(xmlFile);
+
+			IDOMDocument document = domModel.getDocument();
+
+			DocumentTypeImpl docType = (DocumentTypeImpl)document.getDoctype();
+
+			int problemsFixed = 0;
+
+			for (int i = 0; i < problems.size(); i++) {
```

Comment:
@Seiphon  Here this doesn't make any sense, that you are looping over the problems but then performing the exact same set of changes to the model (one for each problem).  So it doesn't matter how many problmes are in the array, you only need to do it once.